SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntityCreationStep.cpp

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "SkeletonEntityCreationStep.h"
+
+#include "EngineClasses/SpatialNetDriver.h"
+#include "Interop/GlobalStateManager.h"
+
+namespace SpatialGDK
+{
+DEFINE_LOG_CATEGORY_STATIC(LogSpatialSkeletonEntityCreationStep, Log, All);
+
+bool FSkeletonEntityCreationStep::TryFinish()
+{
+	if (!bWasInitialized)
+	{
+		bWasInitialized = true;
+		Initialize();
+	}
```

Comment:
Has this been done yet. I'm slightly confused by the initialise function. Why can't it just go in the constructor?

Reply:
The `Initialize` function needs to know about whether this worker has GSM authority to configure the "step" correctly. I don't think we have a well-defined enough place where we could put the creation of this class, so I've picked this as the `TryFinishStartup` calls start getting through after we know who's GSM auth and who isn't.
Now that I think about this, there's the `WellKnownEntitySystem::MaybeClaimSnapshotPartition` so we could create this startup step there? Hmm.

Reply:
Can't we trigger the creation of this from getting authority over the GSM? Or does that cause problems for stopping the stop on non auth servers?

Reply:
Hmm, maybe it would be best to call it something other than initialise but again, naming something isn't the most important thing. We can iterate on this as we change the structure around it.